File: samples/jvm/src/main/java/dev/pubgkt/LeaderboardsSample.java
package dev.pubgkt;

import java.util.List;

import kotlin.coroutines.EmptyCoroutineContext;
import kotlinx.coroutines.BuildersKt;

import dev.pubgkt.leaderboards.GameMode;
import dev.pubgkt.leaderboards.GetLeaderboardKt;
import dev.pubgkt.leaderboards.Leaderboard;

public class LeaderboardsSample {
    public static void main(String[] args) throws InterruptedException {
        if (args.length == 0) {
            throw new IllegalArgumentException("Usage: LeaderboardsSample <api-key>");
        }

        PubgApi api = new PubgApi(args[0]);

        System.out.println("=== getLeaderboard ===");
        List<Season> seasons = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> SeasonKt.seasons(api, Platform.STEAM, cont)
        );
        Season currentSeason = seasons.stream().filter(Season::isCurrentSeason).findFirst().orElseThrow();

        Leaderboard leaderboard = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetLeaderboardKt.getLeaderboard(api, currentSeason.getId(), GameMode.SQUAD_FPP, PlatformRegion.PC_NA, cont)
        );
        System.out.println("Season: " + leaderboard.getSeasonId() + ", Mode: " + leaderboard.getGameMode());
        System.out.println("Top 10:");
        leaderboard.getPlacements().stream().limit(10).forEach(p ->
                System.out.println("  #" + p.getRank() + " " + p.getPlayerName() + " — " + p.getRankPoints() + " RP")
        );
    }
}

File: samples/jvm/src/main/kotlin/dev/pubgkt/LeaderboardsSample.kt
package dev.pubgkt

import dev.pubgkt.leaderboards.GameMode
import dev.pubgkt.leaderboards.getLeaderboard

suspend fun main(args: Array<String>) {
    val apiKey = args.getOrNull(0) ?: error("Usage: LeaderboardsSample <api-key>")
    val api = PubgApi(apiKey)

    println("=== getLeaderboard ===")
    val seasons = api.seasons(Platform.STEAM)
    val currentSeason = seasons.currentOrNull() ?: error("No current season found")

    val leaderboard = api.getLeaderboard(
        seasonId = currentSeason.id,
        gameMode = GameMode.SQUAD_FPP,
        platformRegion = PlatformRegion.PC_NA,
    )
    println("Season: ${leaderboard.seasonId}, Mode: ${leaderboard.gameMode}")
    println("Top 10:")
    leaderboard.placements.take(10).forEach { p ->
        println("  #${p.rank} ${p.playerName} — ${p.rankPoints} RP, ${p.wins} wins, KDA ${p.kda}")
    }
}
